### PR TITLE
[google.accounts] update params for TokenClientConfig and CodeClientConfig in oauth2

### PR DESCRIPTION
**Pull request by codedealer**
Updates the optional parameters for config initialization, including `OverridableTokenClientConfig` available for `TokenClient`.
Outdated arguments were marked as deprecated in the comments.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Google Identity js reference](https://developers.google.com/identity/oauth2/web/reference/js-reference)

### CODE REVIEW

File: types/google.accounts/index.d.ts [hunk 1]
```diff
@@ -45,25 +45,71 @@ declare namespace google.accounts {
...
+             * ignored.
+             *
+             * No effect for newer OAuth client IDs, since more granular
+             * permissions is always enabled for them.
+             */
+            enable_granular_consent?: boolean;
+
+            /**
+             * Deprecated, you should use enable_granular_consent instead.
+             * Optional, defaults to true.
```
Comment (PartyLich) [inline types/google.accounts/index.d.ts:88]: is there a source for this default? Given its deprecation I would imagine it defaults to being unset/undefined/null

File: types/google.accounts/index.d.ts [hunk 2]
```diff
@@ -229,15 +286,40 @@ declare namespace google.accounts {
...
+             * ignored.
+             *
+             * No effect for newer OAuth client IDs, since more granular
+             * permissions is always enabled for them.
+             */
+            enable_granular_consent?: boolean;
+
+            /**
+             * Deprecated, you should use enable_granular_consent instead.
              * Optional, defaults to true.
```
Comment (PartyLich) [inline types/google.accounts/index.d.ts:303]: as above, this default may no longer be the case, and it is not mentioned in the current google documentation.

File: types/google.accounts/index.d.ts [hunk 3]
```diff
@@ -310,6 +412,20 @@ declare namespace google.accounts {
...
+             * ignored.
+             *
+             * No effect for newer OAuth client IDs, since more granular
+             * permissions is always enabled for them.
+             */
+            enable_granular_consent?: boolean;
+
+            /**
+             * Deprecated, you should use enable_granular_consent instead.
              * Optional, defaults to true. If set to false, more granular Google
```
Comment (PartyLich) [inline types/google.accounts/index.d.ts:429]: as above, default value is no longer in the documentation, and may no longer be the case given deprecation

File: types/google.accounts/index.d.ts (hunk 1 above)
Comment (codedealer) [reply types/google.accounts/index.d.ts:88]: Right, when both of these settings aren't set, the optional value of `enable_granular_consent` takes precedence. My rationale was to leave the old description for older codebases, but I will update the comments to match the current Google documentation so there is no confusion.

**Review by PartyLich — APPROVED** [review]
lgtm. thank you for the contrib!

**Comment by PartyLich** [comment]
Ready to merge

**Pull request merged**